examples/Reader.hs

```diff
@@ -17,30 +18,34 @@ import GHC.Generics (Generic)
 import Test.Common
 import Test.Hspec
 
+data Tag = Foo | Bar
```

Comment:
Rather than using `-XDataKind`, how about simply defining

```haskell
data Foo
data Bar
```

So as not to give the impression that `TypeOf` needs to be used with a closed kind?

Reply:
Good point. `-XEmptyDataDecls` it is, then !

Reply:
It's not necessary in recent versions of GHC, right?

Reply:
@mrkkrp This seems like an accurate statement. Is this a Haskell2010 thing?

@gdeest probably the extension line can be removed, then: that's one fewer lines!